packages/compass-indexes/src/components/create-index-form/query-flow-section.tsx

```diff
@@ -166,34 +161,29 @@ const QueryFlowSection = ({
     });
   };
 
-  const handleQueryInputChange = useCallback((text: string) => {
-    setInputQuery(text);
-    setHasNewChanges(true);
-  }, []);
+  const handleQueryInputChange = useCallback(
+    (text: string) => {
+      onQueryUpdated({ query: text, hasQueryChanges: true });
```

Comment:
I mentioned a few times, with redux you really want your reducer and action types to be the driver of the logic, sort of (sorry, it's a bit vague). So the actual driver for this value changing is the fact of the action itself, right? And so reducer can set it to `true` when it's reacting to the query updated action. You can think about this also from this perspective: does it ever make sense to dispatch this action and have `hasQueryChanges` be set to false? As far as I can see it should never be the case, so shouldn't be something that action calls is driving